configuration_helper/models/config.py

```diff
@@ -0,0 +1,53 @@
+# © 2014 David BEAL Akretion
+# © 2016 Yannick Vaucher (Camptocamp SA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import re
+
+from odoo import api, fields, models
+
+
+class AbstractConfigSettings(models.AbstractModel):
+    _name = 'abstract.config.settings'
+    _description = 'Abstract configuration settings'
+    # prefix field name to differentiate fields in company with those in config
+    _prefix = 'setting_'
+    # this is the class name to import in your module
+    # (it should be ResCompany or res_company, depends of your code)
+    _companyObject = None
+    _setup_extra_done = False
+
+    company_id = fields.Many2one(
+        'res.company',
+        'Company',
+        required=True,
+        default=lambda self: self.env.user.company_id
+    )
+
+    def _filter_field(self, field_key):
+        """Inherit in your module to define for which company field
+        you don't want have a matching related field"""
+        return True
+
+    @api.model
+    def _setup_base(self):
+        cls = type(self)
+        super(AbstractConfigSettings, self)._setup_base()
+        if not self._companyObject:
```

Comment:
shouldn't be `cls._companyObject`?

Reply:
``
`-> if not cls._companyObject:
(Pdb) self._companyObject
<class 'odoo.addons.delivery_carrier_label_postlogistics.models.company.ResCompany'>
(Pdb) cls._companyObject
<class 'odoo.addons.delivery_carrier_label_postlogistics.models.company.ResCompany'>
```
Seems to be the same, but makes sense to change it considering `cls._companyObject` is used afterwards.